test/config.js

```diff
@@ -482,6 +483,40 @@ captureBodyTests.forEach(function (captureBodyTest) {
   })
 })
 
+var usePathAsTransactionNameTests = [
+  { value: true, url: '/foo/bar?baz=2', transactionName: 'GET /foo/bar' },
+  { value: false, url: '/foo/bar?baz=2', transactionName: 'GET unknown route' },
+]
+
+usePathAsTransactionNameTests.forEach(function (usePathAsTransactionNameTest) {
+  test('usePathAsTransactionName => ' + usePathAsTransactionNameTest.value, function (t) {
+    t.plan(2)
+
+    var agent = Agent()
+    agent.start({
+      serviceName: 'test',
+      usePathAsTransactionName: usePathAsTransactionNameTest.value
+    })
+
+    var sendTransaction = agent._transport.sendTransaction
+    agent._transport.sendTransaction = function (trans, cb) {
+      t.ok(trans)
+      t.equal(trans.name, usePathAsTransactionNameTest.transactionName)
+      if (cb) process.nextTick(cb)
+    }
+    t.on('end', function () {
+      agent._transport.sendTransaction = sendTransaction
+    })
+
+    var req = new IncomingMessage()
+    req.url = usePathAsTransactionNameTest.url
+    req.method = 'GET'
+
+    var trans = agent.startTransaction()
+    trans.req = req
+    trans.end()
+})
```

Comment:
Seems like you're missing an ending `}` here:

```suggestion
  })
}
```

Reply:
oh, my bad 
fixed :)